File: JANUARY/02-01-2026/961.ex
defmodule Solution do
  @spec repeated_n_times(nums :: [integer]) :: integer
  def repeated_n_times(nums) do
    Enum.reduce_while(nums, MapSet.new(), fn x, set ->
      if MapSet.member?(set, x) do
        {:halt, x}
      else
        {:cont, MapSet.put(set, x)}
      end
    end)
  end
end

File: JANUARY/02-01-2026/961.erl
-spec repeated_n_times(Nums :: [integer()]) -> integer().
repeated_n_times(Nums) ->
    repeated_n_times(Nums, #{ }).

repeated_n_times([], _) ->
    -1;
repeated_n_times([H | T], Set) ->
    case maps:is_key(H, Set) of
        true -> H;
        false -> repeated_n_times(T, maps:put(H, true, Set))
    end.